other/text_to_speech.py

```diff
@@ -0,0 +1,116 @@
+"""
+Created by: Ariel Sperduti
+
+Given a text it converts into an  MP3 audio file.
+Under the hood it uses fromtexttospeech.com.
+
+Example:
+    text_to_speech(text="¡Hola Mundo!", language="Spanish", voice="Mateo")
+"""
+
+import re
+
+import requests
+
+LANGUAGES_VOICES = {
+    "US English": ["Alice", "Daisy", "George", "Jenna", "John"],
+    "British": ["Emma", "Harry"],
+    "French": ["Gabriel", "Jade"],
+    "Spanish": ["Isabella", "Mateo"],
+    "German": ["Michael", "Nadine"],
+    "Italian": ["Alessandra", "Giovanni"],
+    "Portuguese": ["Rodrigo"],
+    "Russian": ["Valentina"],
+}
+
+VOICES = {
+    "Emma": "IVONA Amy22 (UK English)",
+    "Harry": "IVONA Brian22 (UK English)",
+    "Jade": "IVONA CΘline22 (French)",
+    "Gabriel": "IVONA Mathieu22 (French)",
+    "Nadine": "IVONA Marlene22 (German)",
+    "Michael": "IVONA Hans22 (German)",
+    "Valentina": "IVONA Tatyana22 (Russian)",
+    "John": "IVONA Eric22",
+    "Jenna": "IVONA Jennifer22",
+    "George": "IVONA Joey22",
+    "Alice": "IVONA Kimberly22",
+    "Daisy": "IVONA Salli22",
+    "Alessandra": "IVONA Carla22 (Italian)",
+    "Giovanni": "IVONA Giorgio22 (Italian)",
+    "Isabella": "IVONA Conchita22 (Spanish [Modern])",
+    "Mateo": "IVONA Enrique22 (Spanish [Modern])",
+    "Rodrigo": "IVONA Cristiano22 (Portuguese)",
+}
+
+
+def get_mp3_filename(html: str) -> str:
+    """Returns the  mp3 file name of an HTML response from fromtexttospeech.com"""
+    try:
+        filename = re.search(r"(\/output\/\d+\/\d+\.mp3)", html).group(0)
+    except AttributeError:
```

Comment:
Why will there be an `AttributeError` in here?

Also, it seems that the group is unnecessary in the regex as you're getting the entire match as output:
```suggestion
    try:
        filename = re.search(r"\/output\/\d+\/\d+\.mp3", html)
    except AttributeError:
```

Make sure the regex is correct, I might be wrong.

Reply:
The `group` is needed to it, because it returns me the string(without it, I get a Match object), but I can take out the index(in the `group` method) and the `()`(in the regex) because I will take the first occurrence.

Reply:
> Why will there be an `AttributeError` in here?

I put the `AttributeError` because if the `re.search`doesn't found anything I got `None` and `None` doesn't have `group()`
Do you suggest another way?

Reply:
In that case:
```python
if match := re.search(r"\/output\/\d+\/\d+\.mp3", html):
    filename = match.group()
```
You don't actually need to put the parenthesis to get the entire match. Also, if you don't pass any argument to the `group` method, it will assume to be 0 thus returning the entire match.

Reply:
> In that case:
> 
> ```python
> if match := re.search(r"\/output\/\d+\/\d+\.mp3", html):
>     filename = match.group()
> ```

Thanks, I didn't know about the `:=`.
In general, I use the try/except for this: https://docs.python.org/3.9/glossary.html#term-eafp